core/sup_tree_core/periodic_log/writer.ex

```diff
@@ -0,0 +1,83 @@
+# Copyright(c) 2015-2019 ACCESS CO., LTD. All rights reserved.
+
+use Croma
+
+defmodule AntikytheraCore.PeriodicLog.Writer do
+  @moduledoc """
+  A `GenServer` that logs periodically
+  """
+
+  use GenServer
+  alias AntikytheraCore.GearLog.FileHandle
+  alias Antikythera.{Time, ContextId}
+
+  @interval        1000
+  @rotate_interval 7_200_000
+
+  defmodule State do
+    use Croma.Struct, recursive_new?: true, fields: [
+      file_handle: Croma.Tuple, # FileHandle.t
+      empty?:      Croma.Boolean,
+      timer:       Croma.Reference,
+      build_mod:   Croma.Atom,
```

Comment:
I think "mod" (in `build_mod:`) is ambiguous because it can stand for things other than module (though it becomes obvious that "mod" stands for module after reading the code).
I prefer to use "module" here.
In my opinion, using an abbreviation for broadly used name (such as Struct’s attributes, arguments of public function, …) leads to a bit of confusion.

Reply:
In Elixir world, it seems to be generally accepted that `mod:` stands for module.
I need to study more...
https://elixir-lang.org/getting-started/mix-otp/supervisor-and-application.html#the-application-callback